solutions/scripts/target.js

```diff
@@ -0,0 +1,145 @@
+import { getMetadata, sampleRUM } from './lib-franklin.js';
+
+const ADOBE_TARGET_SESSION_ID_PARAM = 'adobeTargetSessionId';
+
+/**
+ * Convert a URL to a relative URL.
+ * @param url
+ * @returns {*|string}
+ */
+function toRelativeUrl(url) {
+  try {
+    const parsedUrl = new URL(url);
+    return parsedUrl.pathname + parsedUrl.search + parsedUrl.hash;
+  } catch (e) {
+    return url;
+  }
+}
+
+/**
+ * Generate a random session id.
+ * @param length
+ * @returns {string}
+ */
+function generateSessionID(length = 16) {
+  const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+  let sessionID = '';
+  for (let i = 0; i < length; i += 1) {
+    const randomIndex = Math.floor(Math.random() * characters.length);
+    sessionID += characters.charAt(randomIndex);
+  }
+  return sessionID;
+}
+
+/**
+ * Get or create a session id for the current user.
+ * @returns {string}
+ */
+function getOrCreateSessionId() {
+  let sessionId = sessionStorage.getItem(ADOBE_TARGET_SESSION_ID_PARAM);
+  // eslint-disable-next-line no-console
+  console.debug(`Session id: ${sessionId}`);
+  if (!sessionId) {
+    sessionId = generateSessionID();
+    // eslint-disable-next-line no-console
+    console.debug(`Generated new session id: ${sessionId}`);
+    sessionStorage.setItem(ADOBE_TARGET_SESSION_ID_PARAM, sessionId);
+  }
+  return sessionId;
+}
+
+/**
+ * Fetch the target offers for the current location.
+ * @returns {Promise<boolean>}
+ */
+async function fetchChallengerPageUrl(tenant, targetLocation) {
+  // eslint-disable-next-line no-console
+  console.debug(`Fetching target offers for location: ${targetLocation}`);
+  const res = await fetch(`https://${tenant}.tt.omtrdc.net/rest/v1/delivery?client=${tenant}&sessionId=${getOrCreateSessionId()}`, {
```

Comment:
No sure I fully understand this. Could please you elaborate?

Reply:
You can usually specify a "prod domain" for the endpoint in alloy, so for instance you could map this to `https://target.bitdefender.com` so cookies are on the same domain for instance, or just `https://www.bitdefender.com` to avoid resolving another host (save the DNS+TLS handshake to save ~500ms).

Since you do hardcode the domain here, this will not be an option. So you need 3rd-party cookies and another DNS/TLS handshake… which might be ok or not :)

Reply:
You mean the `edgeDomain` setting that needs the CNAME record changed, right?

Reply:
yes, that's the one